link/src/main/java/com/stripe/android/link/ui/paymentmethod/PaymentMethodViewModel.kt

```diff
@@ -202,15 +204,33 @@ internal class PaymentMethodViewModel @Inject constructor(
         navigator.cancel(reason = PayAnotherWay)
     }
 
-    private fun completePayment(linkPaymentDetails: LinkPaymentDetails) {
-        val factory = ConfirmStripeIntentParamsFactory.createFactory(
-            clientSecret = requireNotNull(stripeIntent.clientSecret),
-            shipping = args.shippingValues?.toConfirmPaymentIntentShipping(),
+    private suspend fun completePayment(linkPaymentDetails: LinkPaymentDetails) {
+        val nextStep = intentConfirmationInterceptor.intercept(
+            clientSecret = stripeIntent.clientSecret,
+            paymentMethodCreateParams = linkPaymentDetails.paymentMethodCreateParams,
+            shippingValues = args.shippingValues?.toConfirmPaymentIntentShipping(),
+            setupForFutureUsage = null,
         )
 
-        val params = factory.create(linkPaymentDetails.paymentMethodCreateParams)
+        when (nextStep) {
+            is IntentConfirmationInterceptor.NextStep.Confirm -> {
+                confirmStripeIntent(nextStep.confirmParams)
+            }
+            is IntentConfirmationInterceptor.NextStep.HandleNextAction -> {
+                // This can't happen for Link
```

Comment:
Is 3DS not enabled for Link?

Reply:
Should we throw an error here? If this changes in the future, it would be nice to know about it.

Reply:
It was my assumption that 3DS isn’t supported because [the 3DS test numbers](https://stripe.com/docs/testing#regulatory-cards) don’t work in Link. I’m following up with the Link team to figure out what’s correct here. Thanks for pushing back on this.

Reply:
3DS is not supported, but 3DS2 is. I’ll add handling for this.